lib/monaco-adapter.js

```diff
@@ -66,34 +67,60 @@ firepad.MonacoAdapter = (function () {
     }, 0);
   };
 
-  MonacoAdapter.prototype.operationFromMonacoChange = function (change, offsetLength) {
+  MonacoAdapter.prototype.operationFromMonacoChange = function (change, content, offsetLength) {
     text = change.text;
-    start = this.indexFromPos(change.range, 'start');
-    restLength = this.lastDocLines.join(this.monacoModel.getEOL()).length - start + offsetLength;
+    start = change.rangeOffset + offsetLength;
+
+    restLength = content.length - start + offsetLength;
     text_ins = change.text;
-    rangeLen = change.rangeLength;
+    rangeLength = change.rangeLength;
 
-    if (change.rangeLength > 0) {
-      restLength -= rangeLen;
-      text = this.lastDocLines.join(this.monacoModel.getEOL())
-        .slice(start, start + rangeLen);
-    }
+    if (text.length === 0 && rangeLength > 0) {
+      // Delete operation.
+      var replacedText = content.slice(start, start + rangeLength);
+
+      var delete_op = new firepad.TextOperation()
+          .retain(start)
+          ['delete'](rangeLength)
```

Comment:
This is pretty strange syntax, why do we have to access `delete` this way but we access `insert()` as a method?  Sorry if this is a noob question I am new-ish to this codebase.

Reply:
Seems that below we don't do this:

```
      var inverse_op = new firepad.TextOperation()
          .retain(start)
          .delete(text)
          .retain(restLength);
```

Reply:
Done. (Pretty sure I just cargo-cultishly copied that...)